snow/validators/gvalidators/validator_state_test.go

```diff
@@ -181,3 +182,49 @@ func TestGetValidatorSet(t *testing.T) {
 	_, err = state.client.GetValidatorSet(context.Background(), height, subnetID)
 	require.Error(err)
 }
+
+// BenchmarkGetValidatorSet measures the time it takes to complete a gRPC client
+// request based on a mocked validator set.
+func BenchmarkGetValidatorSet(b *testing.B) {
+	for _, i := range []int{1, 10, 1000, 2000} {
+		vs := setupValidatorSet(b, i)
+		b.Run(fmt.Sprintf("get_validator_set_%d_validators", i), func(b *testing.B) {
+			benchmarkGetValidatorSet(b, vs)
+		})
+	}
+}
+
+func benchmarkGetValidatorSet(b *testing.B, vs map[ids.NodeID]*validators.GetValidatorOutput) {
+	require := require.New(b)
+	ctrl := gomock.NewController(b)
+	state := setupState(b, ctrl)
+	defer func() {
+		ctrl.Finish()
+		state.closeFn()
+	}()
+
+	height := uint64(1337)
+	subnetID := ids.GenerateTestID()
+	for i := 0; i < b.N; i++ {
+		state.server.EXPECT().GetValidatorSet(gomock.Any(), height, subnetID).Return(vs, nil)
+		_, err := state.client.GetValidatorSet(context.Background(), height, subnetID)
+		require.NoError(err)
+	}
+}
+
+func setupValidatorSet(b *testing.B, size int) map[ids.NodeID]*validators.GetValidatorOutput {
+	b.Helper()
+
+	set := make(map[ids.NodeID]*validators.GetValidatorOutput, size)
+	sk, _ := bls.NewSecretKey()
```

Comment:
We should either not take in `b`, or we should use it somewhere in this function... Generally I think it's good to handle the errors (even in tests) when possible
```suggestion
	sk, err := bls.NewSecretKey()
	require.NoError(b, err)
```